synapse/util/threepids.py

```diff
@@ -48,3 +48,25 @@ def check_3pid_allowed(hs, medium, address):
         return True
 
     return False
+
+
+def check_3pid_valid_format(medium, address):
+    """Checks whether 3pid has a valid format
+
+    Args:
+        medium (str): 3pid medium - e.g. email, msisdn
+        address (str): 3pid address to check (e.g. "wotan@matrix.org")
+    Returns:
+        bool: whether the email address has a valid format
+    """
+
+    if medium == "email":
+        regex = r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
```

Comment:
At a glance this looks like it'll hit most valid e-mails, but usually things like e-mails and domains are relatively hard to validate -- where did this regular expression come from?

Reply:
Source of regex is: https://emailregex.com/

> Python
 r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"

We can change the regex. That is the reason why the validation is a new function.

Reply:
I did not found a good regex for msidns.

Reply:
The main issue I see here is that this doesn't support international domain names (or Unicode in general, although I'm not sure if Unicode is valid in the local part of an email address).

Taking a look at that site, it seems the Python regular expression is *much* simpler than some of the others, which I find concerning.

Poking around a bit, [some other](https://stackoverflow.com/questions/8022530/how-to-check-for-valid-email-address) solutions seem to suggest being generous since it is hard to know the form of an email without trying to actually validate it and use something like: `r"^[^@]+@[^@]+\.[^@]+$"`.